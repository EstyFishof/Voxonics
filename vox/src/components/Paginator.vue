<template>
  <div class="paginator-main">
    <div class="paginator">
      <div class="select item-container">

        <div v-click-outside="hideListPerPage" @click="showPerPage = !showPerPage" content="Quantity per page"
             v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }"
             class="left-item peerpage-wrap">
          <label>
            <svg class="list" width="19" height="19" viewBox="0 -3 19 15" fill="none" xmlns="http://www.w3.org/2000/svg">
              <rect x="4.76953" y="0.5" width="14.2305" height="1.76923" rx="0.884615"/>
              <rect x="4.76953" y="6.26953" width="14.2305" height="1.76923" rx="0.884615"/>
              <rect x="4.76953" y="12.0391" width="14.2305" height="1.76923" rx="0.884615"/>
              <circle cx="1.5" cy="1.5" r="1.5"/>
              <circle cx="1.5" cy="7.19922" r="1.5"/>
              <circle cx="1.5" cy="13" r="1.5"/>
            </svg>
            <span class="per-page">{{ this['per-page'] }}</span>
            <span class="triangle" :class="{'rotate-arrow': showPerPage}">
              <svg width="14" height="7" viewBox="0 0 14 7" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path d="M1 0.999999L7 6L13 1" stroke="var(--paginator-arrow-list)" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
              </svg>
            </span>
            <ul v-show="showPerPage" class="peerpage-select">
              <li @click="setPeerPage('12')">12</li>
              <li @click="setPeerPage('24')">24</li>
              <li @click="setPeerPage('36')">36</li>
              <li @click="setPeerPage('48')">48</li>
            </ul>
          </label>
        </div>
        <div class="pages">
          <div @click="goToStart()" content="First page"
               v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }" class="arrow-wrap">
            <svg :class="{disabled: +first === 1}" class="arrow left" width="30"
                 height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
            <svg :class="{disabled: +first === 1}" class="arrow left" width="30"
                 height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
          </div>
          <div content="Previous page" v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }"
               class="arrow-wrap">
            <svg @click.stop="leafThrough('back')" :class="{disabled: +first === 1}" class="arrow left" width="30"
                 height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
          </div>
          <div class="first-page peerpage-wrap">
            <span>{{ paginator.currentPage - 1 || '' }}</span>
            <span v-show="paginator.currentPage - 1">,</span>
            <span class="active">{{ paginator.currentPage }}</span>
            <span v-show="lastPage - paginator.currentPage > 1">...</span>
            <span v-show="paginator.currentPage === lastPage - 1">,</span>
            <span v-show="lastPage !== paginator.currentPage"
                  :class="{active: lastPage === paginator.currentPage}">{{ lastPage }}</span>
          </div>
          <div content="Next page" v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }"
               class="arrow-wrap">
            <svg @click.stop="leafThrough('forward')" :class="{disabled: +last === +paginator.count || !paginator.count}"
                 class="arrow right"
                 width="30" height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
          </div>
          <div @click="goToEnd()" content="Last page"
               v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }" class="arrow-wrap">
            <svg :class="{disabled: +last === +paginator.count}"
                 class="arrow right"
                 width="30" height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
            <svg :class="{disabled: +last === +paginator.count}"
                 class="arrow right"
                 width="30" height="18" viewBox="0 0 10 18" fill="transparent"
                 xmlns="http://www.w3.org/2000/svg">
              <path d="M9 1L1 9L9 17" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
            </svg>
          </div>
        </div>
        <div class="right-item peerpage-wrap search-dep">
          <svg class="search" width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="circle" opacity="0.7" fill-rule="evenodd" clip-rule="evenodd"
                  d="M1.77843 7.11112C1.77843 4.16979 4.17044 1.77778 7.11177 1.77778C10.0531 1.77778 12.4451 4.16979 12.4451 7.11112C12.4451 10.0525 10.0531 12.4445 7.11177 12.4445C4.17044 12.4445 1.77843 10.0525 1.77843 7.11112ZM15.7396 14.4827L12.7218 11.464C13.6578 10.2605 14.2222 8.75202 14.2222 7.11112C14.2222 3.19023 11.032 0 7.11112 0C3.19023 0 0 3.19023 0 7.11112C0 11.032 3.19023 14.2222 7.11112 14.2222C8.75202 14.2222 10.2605 13.6578 11.464 12.7218L14.4827 15.7396C14.656 15.9129 14.8836 16 15.1111 16C15.3387 16 15.5662 15.9129 15.7396 15.7396C16.0871 15.392 16.0871 14.8302 15.7396 14.4827Z"
            />
            <path d="M5.28784 9.99414L10.995 4.28697L11.9165 5.20842L6.20929 10.9156L5.28784 9.99414Z" fill="#00C82C"
                  fill-opacity="0.6"/>
            <path d="M3 7.70703L3.82635 6.88068L6.11381 9.16814L5.28746 9.99449L3 7.70703Z" fill="#00C82C"
                  fill-opacity="0.6"/>
          </svg>

          <span class="count">{{ paginator.count }}</span>
        </div>
      </div>
    </div>

  </div>
</template>

<script>


import ClickOutside from "vue-click-outside"

export default {
  name: "Paginator",
  props: ['paginator', 'methods', 'perPage'],
  data() {
    return {
      'per-page': this.perPage ?? 12,
      page: 1,
      showPerPage: false
    }
  },
  watch: {
    paginator() {
      this['per-page'] = JSON.parse(JSON.stringify(this.paginator.recordsPerPage))
    },
  },
  computed: {
    lastPage() {
      const lastPageNumber = new URLSearchParams(this.paginator.lastPageUrl).get('page')
      if(this.paginator.currentPage > +lastPageNumber) {
        const method = this.methods.page
        this.$store.commit(method, this.paginator.currentPage - 1)
      }
      return (+lastPageNumber) || ''
    },
    first: function () {
      return (this.paginator.currentPage * this.paginator.recordsPerPage - this.paginator.recordsPerPage) || 1
    },
    last: function () {
      let count = ((this.paginator.currentPage + 1) * this.paginator.recordsPerPage - this.paginator.recordsPerPage)
      if (count >= this.paginator.count) {
        return this.paginator.count || 1
      } else {
        return count || 1
      }
    }
  },
  methods: {
    leafThrough(direction) {
      if (direction === 'back') {
        this.page -= 1
      } else {
        this.page += 1
      }
      const method = this.methods.page
      this.$store.commit(method, this.page)
      return this.page
    },
    goToEnd() {
      const method = this.methods.page
      this.page = Math.ceil(this.paginator.count / this['per-page'])
      this.$store.commit(method, this.page)
    },
    goToStart() {
      const method = this.methods.page
      this.page = 1
      this.$store.commit(method, this.page)
    },
    setPeerPage(value) {
      const method = this.methods.peerPage
      this['per-page'] = value
      this.$store.commit(method, this['per-page'])
      return this['per-page']
    },
    hideListPerPage() {
      this.showPerPage = false
    }
  },
  mounted() {
    this.popupItem = this.$el
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss">
.paginator-main {
  display: flex;
  justify-content: flex-end;
}

.count {
  color: #fff;
  font-size: 16px;
}

.paginator {
  color: #646D75;
  padding: 13px 5px;
  background-color: var(--paginator-bg);
  border-radius: 8px;
  max-height: 48px;
  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

  .left-item {
    margin-right: 8px;

    label {
      position: relative;

      svg {
        filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
      }

      ul {
        position: absolute;
        top: 23px;
        right: 9px;
        z-index: 999;

        li {
          background-color: var(--paginator-item);
          padding: 5px 0 5px 7px;
          width: 53px;

          &:hover {
            background-color: var(--light-blue);
            color: white;
          }
        }
      }
    }
  }

  .right-item {
    margin-left: 8px;
  }

  .active {
    color: var(--light-blue) !important;
  }

  .select {
    height: 100%;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-items: center;
    font-size: 18px;
    font-weight: 300;
    box-shadow: none !important;
    border-radius: 2px;

    .first-page {
      margin-right: 4px;
    }

    .peerpage-wrap {
      background-color: var(--paginator-item);
      height: 22px;
      border-radius: 2px;
      padding: 2px;
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-between;

      .per-page {
        margin-left: 8px;
        margin-right: 5px;
      }

      .triangle {
        svg {
          transform: rotate(-90deg);
          margin-bottom: 2px;
          transition: all .3s ease-in-out;
        }

        &.rotate-arrow {
          svg {
            transform: rotate(0deg);
            margin-bottom: 2px;
            transition: all .3s ease-in-out;
          }
        }
      }

      .peerpage-select {
        margin-top: 0;
        margin-left: 5px;
        width: 45px;
        -webkit-appearance: none;
        -moz-appearance: none;
        background-image: url(../assets/img/icons/arrow-down-white.svg);
        background-position: 93% 50%;
        background-repeat: no-repeat;
        background-size: 16px;
        padding-left: 3px;
        box-shadow: none;
        background-color: transparent;
        color: var(--paginator-color);
        font-size: 18px;
      }

      .search {
        margin-right: 3px;
      }

      span {
        margin: 0 1px;
        color: var(--paginator-color);
      }
    }

    .pages {
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      align-items: center;

      .arrow-wrap:nth-child(5) {
        margin-right: 0;
      }

      .arrow-wrap {
        width: 22px;
        height: 22px;
        background-color: var(--paginator-item);
        margin-right: 4px;
        display: flex;
        align-items: center;
        justify-content: center;
        box-shadow: var(--paginator-shadow);
        border-radius: 2px;
        padding: 2px;

        svg.arrow {
          stroke: #276899;
          fill: transparent;
          text-align: right;

          &.right {
            transform: rotate(180deg);
          }

          &.disabled {
            stroke: var(--paginator-arrow-disabled) !important;
            pointer-events: none;
          }
        }
      }
    }
  }

  .item-container {
    box-shadow: none;
  }
}

.peerpage-wrap {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2px;
  height: 22px;
  box-shadow: var(--paginator-shadow);

  &.search-dep {
    svg {
      filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
    }
  }
}

.circle {
  fill: var(--light-blue);
}

.list rect,
.list circle {
  fill: var(--paginator-color);
}
</style>
