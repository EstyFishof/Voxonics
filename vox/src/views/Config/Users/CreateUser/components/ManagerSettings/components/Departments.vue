<template>
  <section class="table-container" v-if="getRole !== 'admin' && getRole !== 'owner'">
    <div class="table-title">Departments to be managed</div>
    <div class="wrap-table">
      <ul class="table-header">
        <li class="header-item" @click="sortFunc('name', 'depList', 'name')">
          Name of Department
          <img ref="name" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
        <li class="header-item-action" @click="sortFunc('edit', 'depList', 'action')">
          Actions
          <img ref="action" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
      </ul>
      <div>
        <div class="list-group" :class="{'add-shadow': depList.length > 4}" @scroll="funScroll($event)">
          <ul class="list-group-item" v-for="dep in depList" :key="dep.uid.slice(0, 10)">
            <li class="department-name ellipsis">{{ dep.name }}</li>
            <li class="action">
              <label class="label-container">
                <input @change.prevent="chooseDep(dep)" class="view" :checked="dep.view" type="checkbox"/>
                <span class="check"></span>
                <span class="text">View</span></label>
              <label class="label-container">
                <input @change.prevent="setEdit(dep)" class="edit" :checked="dep.edit" type="checkbox"/>
                <span class="check"></span>
                <span class="text">Edit</span></label>
            </li>
          </ul>
        </div>
      </div>
    </div>

  </section>
</template>

<script>
// import {DepartmentGettersApi} from '@/API/getters'
import { mapGetters } from "vuex"
import { serverSort, clientSort } from '@/lib/sort.js'

export default {
  name: "Departments",
  data() {
    return {
      depList: [],
      selectedDepList: [],
      order: {
        name: 'asc',
        edit: 'asc'
      },
      activeSort: 'name',
      count: 10,
      perPage: 9
    }
  },
  watch: {
    getRelatedDepartments() {
      if (this.getDepManaged.length > 0) {
        this.selectedDepList = this.getDepManaged
      }
      this.createDataForView()
    },
    getDepManaged() {
      this.createDataForView()
    }
  },
  computed: {
    ...mapGetters(['getRelatedDepartments', 'getDepManaged', 'getRole', 'getForm', 'getPaginatorDepartment'])
  },
  methods: {
    funScroll({target}) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        if (this.getPaginatorDepartment.paginator.count > this.perPage) {
          this.perPage += 9
          this.fetch(this.perPage, this.activeSort, this.order[this.activeSort] === 'desc' ? 'asc' : 'desc')
        }
      }
    },
    fetch(peerPage, sort, order) {
      this.$store.dispatch('setGlobalDepartmentsList', {peerPage, sort, order})
    },
    sortFunc(param, arrayParam, refElement) {
      if (param !== 'edit') {
        const dataSort = serverSort(this.fetch, this.$refs[refElement], this.order[param], this.perPage, param)
        this.order[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
        this.activeSort = param
      } else {
        const dataSort = clientSort(param, this[arrayParam], this.$refs[refElement], this.order[param])
        this[arrayParam] = dataSort.array
        this.order[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
      }
    },
    chooseDep(dep) {
      const includes = this.selectedDepList.filter(d => d.uid === dep.uid)
      if (!includes.length) {
        dep.view = true
        this.selectedDepList.push(dep)
      } else {
        this.selectedDepList = this.selectedDepList.filter(q => q.uid !== dep.uid)
        dep.edit = false
        this.depList.forEach((de, i) => {
          if (dep.uid === de.uid) {
            de.edit = false
            //Vue2 не видит изменений этих полей, поэтому подменяем сам объект
            const dd = {...de}
            this.depList.splice(i, 1, dd)
          }
        })
      }
      this.$store.dispatch('depManagerSelected', this.selectedDepList)
    },
    setEdit(dep) {
      const department = Object.assign({}, dep)
      if (!department.view) {
        department.view = true
        department.edit = true
        this.depList.forEach((de, i) => {
          if (department.uid === de.uid) {
            de.view = true
            de.edit = true
            //Vue2 не видит изменений этих полей, поэтому подменяем сам объект
            const dd = {...de}
            this.depList.splice(i, 1, dd)
          }
        })
      } else {
        department.edit = !department.edit
      }
      this.selectedDepList = this.selectedDepList.filter(q => q.uid !== department.uid)
      this.selectedDepList.push(department)
      this.$store.dispatch('depManagerSelected', this.selectedDepList)
    },
    createDataForView() {
      this.depList = []
      this.getRelatedDepartments.map(dep => {
        dep.view = false
        dep.edit = false
        let f = false
        if (this.getDepManaged.length > 0) {
          this.getDepManaged.forEach(item => {
            let d = JSON.parse(JSON.stringify(item))
            if (d.uid === dep.uid) {
              f = true
              d.view = true
              if (d.edit) d.view = true
              this.depList.push(d)
            }
          })
          if (!f) {
            this.depList.push(dep)
          }
        } else {
          this.depList.push(dep)
        }
      })
    }
  },
  mounted() {
    //Загружаем выбраный массив из стора
    if (this.getDepManaged.length > 0) {
      this.selectedDepList = this.getDepManaged
    }
    //если есть данные в сторе, применяем их для отображения
    this.createDataForView()
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/users/AvailableQueues.scss"></style>
<style scoped lang="scss">
.table-container {
  .wrap-table {
    .table-header {
      justify-content: space-between;
    }

    .header-item {
      width: 260px;
    }

    .header-item-action {
      padding-left: 0;
      width: 150px;
    }

    .list-group-item {
      justify-content: space-between;

      .action {
        width: 150px;
      }
    }

    .department-name {
      width: 220px;
      text-align: left;
    }
  }
}
</style>
