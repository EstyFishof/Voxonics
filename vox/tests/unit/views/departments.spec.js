import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Departments from '../../../src/views/Config/Departments/Departments.vue';
import Paginator from '../../../src/components/Paginator.vue';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
const router = new VueRouter();
localVue.use(Vuex);

describe('Departments', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const store = new Vuex.Store({
        state: {
            paginator: {peerPage: 10, page: 1, paginator: {count: 50, recordsPerPage: 5}},
            globalSearch: ''
        },
        actions: {
            setStrategy({commit}, value) {},
            setTimeout({commit}, value) {},
            setRinginuse({commit}, value) {},
            setWait({commit}, value) {},
        },
        mutations: {
            managersQueueInfoMutation(state, value) {
                state.listManagers = value;
            }
        },
        getters: {
            getGlobalSearch(state) {
                return state.globalSearch;
            },
            getPaginatorDepartment(state) {
                return state.paginator;
            }
        }
    });

    // const mock = jasmine.createSpy()
    const mock = jest.fn();

    const StubbedComponent = {
        render() {},
        paginator: {
            setDataPeerPage: () => jest.fn()
        }
    };
    const wrapper = shallowMount(Departments, { store, localVue, router, stubs: {Departments: true, paginator: Paginator}, mocks: {$refs: {paginator: {setDataPeerPage: () => jest.fn()}}} });
    // wrapper.vm.$refs.stubbedComponent.mock()



    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('onscroll toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'onscroll');
        await wrapper.vm.onscroll();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteScroll toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteScroll');
        await wrapper.vm.deleteScroll();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('funScroll toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'funScroll');
        await wrapper.vm.funScroll();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('funScroll recordsPerPage < count', async () => {
        // await wrapper.setData({paginator: {recordsPerPage: 9, count: 12}})
        const spy = jest.spyOn(wrapper.vm, 'funScroll');
        await wrapper.vm.funScroll();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('viewDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'viewDepartment');
        await wrapper.vm.viewDepartment({name: 'dep', uid: 5000});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('goToAddNewUserPage toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'goToAddNewUserPage');
        await wrapper.vm.goToAddNewUserPage();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('name');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('name');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('fetch toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'fetch');
        await wrapper.vm.fetch('url', 9, 'name', 9, 'name');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('created component', () => {
        Departments.created.call(wrapper.vm);
    });
    it('mounted component', async () => {
        // console.log({'1111' : wrapper.vm.$refs});0
        // wrapper.$refs
        // await store.commit('managersQueueInfoMutation', [{}, {}])
        await Departments.mounted.call(wrapper.vm);
    });
    it('beforeDestroy component', () => {
        // console.log({'1111' : wrapper.vm.$refs});
        Departments.beforeDestroy.call(wrapper.vm);
    });

});
