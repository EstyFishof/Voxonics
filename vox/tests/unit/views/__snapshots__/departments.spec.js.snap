// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Departments Check snapshot 1`] = `
<div
  class="get-users-page-container"
>
  <div
    class="toolbar"
  >
    <div
      class="bar-wrapper"
    >
      <div
        class="btn-img-department"
      >
        <span
          class="add-btn-dep"
          content="Add new department"
          tabindex="0"
        >
          <img
            alt="plus"
            class="add-btn-plus"
            src="@/assets/img/icons/plus.svg"
            width="12"
          />
           
          <img
            alt="plus"
            class="add-btn-plus department-icon"
            height="29"
            src="@/assets/img/icons/department.svg"
            width="30"
          />
        </span>
         
        <div
          class="btn-img"
        >
          <svg
            class="switch-view switch-list"
            content="List mode"
            fill="none"
            height="16"
            tabindex="0"
            viewBox="0 0 19 15"
            width="23"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="1.76923"
              rx="0.884615"
              width="14.2305"
              x="4.76953"
              y="0.5"
            />
             
            <rect
              height="1.76923"
              rx="0.884615"
              width="14.2305"
              x="4.76953"
              y="6.26953"
            />
             
            <rect
              height="1.76923"
              rx="0.884615"
              width="14.2305"
              x="4.76953"
              y="12.0391"
            />
             
            <circle
              cx="1.5"
              cy="1.5"
              r="1.5"
            />
             
            <circle
              cx="1.5"
              cy="7.19922"
              r="1.5"
            />
             
            <circle
              cx="1.5"
              cy="13"
              r="1.5"
            />
          </svg>
        </div>
      </div>
       
      <ul
        class="sort-tabs"
      >
        <li>
          Filter by:
        </li>
         
        <li>
          <label>
            <input
              class="input-field"
              placeholder="Name/Number"
              type="text"
            />
          </label>
        </li>
      </ul>
    </div>
     
    <div
      class="paginator-main"
    >
      <div
        class="paginator"
      >
        <div
          class="select item-container"
        >
          <div
            class="left-item peerpage-wrap"
            content="Quantity per page"
            tabindex="0"
          >
            <label>
              <svg
                class="list"
                fill="none"
                height="15"
                viewBox="0 0 19 15"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="1.76923"
                  rx="0.884615"
                  width="14.2305"
                  x="4.76953"
                  y="0.5"
                />
                 
                <rect
                  height="1.76923"
                  rx="0.884615"
                  width="14.2305"
                  x="4.76953"
                  y="6.26953"
                />
                 
                <rect
                  height="1.76923"
                  rx="0.884615"
                  width="14.2305"
                  x="4.76953"
                  y="12.0391"
                />
                 
                <circle
                  cx="1.5"
                  cy="1.5"
                  r="1.5"
                />
                 
                <circle
                  cx="1.5"
                  cy="7.19922"
                  r="1.5"
                />
                 
                <circle
                  cx="1.5"
                  cy="13"
                  r="1.5"
                />
              </svg>
               
              <span
                class="per-page"
              >
                10
              </span>
               
              <span
                class="triangle"
              >
                <img
                  alt="select"
                  src="@/assets/img/icons/arrow-down-white.svg"
                />
              </span>
               
              <ul
                class="peerpage-select"
                style="display: none;"
              >
                <li>
                  12
                </li>
                 
                <li>
                  24
                </li>
                 
                <li>
                  36
                </li>
                 
                <li>
                  48
                </li>
              </ul>
            </label>
          </div>
           
          <div
            class="pages"
          >
            <div
              class="arrow-wrap"
              content="First page"
              tabindex="0"
            >
              <svg
                class="arrow left disabled"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
               
              <svg
                class="arrow left disabled"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
             
            <div
              class="arrow-wrap"
              content="Previous page"
              tabindex="0"
            >
              <svg
                class="arrow left disabled"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
             
            <div
              class="first-page peerpage-wrap"
            >
              <span>
                
              </span>
               
              <span
                style="display: none;"
              >
                ,
              </span>
               
              <span
                class="active"
              >
                
              </span>
               
              <span
                style="display: none;"
              >
                ...
              </span>
               
              <span
                style="display: none;"
              >
                ,
              </span>
               
              <span
                class=""
              >
                
              </span>
            </div>
             
            <div
              class="arrow-wrap"
              content="Next page"
              tabindex="0"
            >
              <svg
                class="arrow right"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
             
            <div
              class="arrow-wrap"
              content="Last page"
              tabindex="0"
            >
              <svg
                class="arrow right"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
               
              <svg
                class="arrow right"
                fill="transparent"
                height="18"
                viewBox="0 0 10 18"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L1 9L9 17"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
           
          <div
            class="right-item peerpage-wrap search-dep"
          >
            <svg
              class="search"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="circle"
                clip-rule="evenodd"
                d="M1.77843 7.11112C1.77843 4.16979 4.17044 1.77778 7.11177 1.77778C10.0531 1.77778 12.4451 4.16979 12.4451 7.11112C12.4451 10.0525 10.0531 12.4445 7.11177 12.4445C4.17044 12.4445 1.77843 10.0525 1.77843 7.11112ZM15.7396 14.4827L12.7218 11.464C13.6578 10.2605 14.2222 8.75202 14.2222 7.11112C14.2222 3.19023 11.032 0 7.11112 0C3.19023 0 0 3.19023 0 7.11112C0 11.032 3.19023 14.2222 7.11112 14.2222C8.75202 14.2222 10.2605 13.6578 11.464 12.7218L14.4827 15.7396C14.656 15.9129 14.8836 16 15.1111 16C15.3387 16 15.5662 15.9129 15.7396 15.7396C16.0871 15.392 16.0871 14.8302 15.7396 14.4827Z"
                fill-rule="evenodd"
                opacity="0.7"
              />
               
              <path
                d="M5.28784 9.99414L10.995 4.28697L11.9165 5.20842L6.20929 10.9156L5.28784 9.99414Z"
                fill="#00C82C"
                fill-opacity="0.6"
              />
               
              <path
                d="M3 7.70703L3.82635 6.88068L6.11381 9.16814L5.28746 9.99449L3 7.70703Z"
                fill="#00C82C"
                fill-opacity="0.6"
              />
            </svg>
             
            <span
              class="count"
            >
              50
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
   
  <div
    class="data-container"
  >
    <!---->
  </div>
</div>
`;
